### PR TITLE
docs: Add API documentation

### PR DESCRIPTION
**Pull request by mikemckiernan**
* Set the Sphinx version to 3.5.4 to sidestep
  an impending stylesheet issue with the
  the CSS that docs.nvidia.com prefers.

### CODE REVIEW

File: docs/source/api.rst [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
... [autosummary] failed to import 'merlin.models.tf.blocks.aggregation.SequenceAggregation': no module named merlin.models.tf.blocks.aggregation.SequenceAggregation                                                                                                                                                    WARNING: [autosummary] failed to import 'merlin.models.tf.blocks.aggregation.SequenceAggregator': no module named merlin.models.tf.blocks.aggregation.SequenceAggregator                                                                                                                                                      WARNING: [autosummary] failed to import 'merlin.models.tf.blocks.aggregation.StackFeatures': no module named merlin.models.tf.blocks.aggregation.StackFeatures WARNING: [autosummary] failed to import 'merlin.models.tf.blocks.aggregation.Sum': no module named merlin.models.tf.blocks.aggregation.Sum                     WARNING: [autosummary] failed to import 'merlin.models.tf.blocks.aggregation.SumResidual': no module named merlin.models.tf.blocks.aggregation.SumResidual     WARNING: [autosummary] failed to import 'merlin.models.tf.blocks.aggregation.TupleAggregation':
```
Comment (benfred) [inline docs/source/api.rst:6]: Are these warnings about failing to import supposed to be here?

File: docs/source/conf.py [hunk 2]
```diff
@@ -93,3 +93,11 @@
 }
 
 autodoc_inherit_docstrings = False
+autodoc_default_options = {
+    "members": True,
+    "undoc-members": True,
+    "show-inheritance": False,
+    "member-order": "bysource",
+}
```
Comment (benfred) [inline docs/source/conf.py:101]: nice! I think we should add this to all of our projects

**Comment by mikemckiernan** [comment]
The pytorch failure is a mystery and not in a fun, delightful way.

The API doc seems to work again: https://nvidia-merlin.github.io/models/review/pr-212/api.html

**Comment by karlhigley** [comment]
Pretty sure that's just a flaky test

**Review by benfred — APPROVED** [review]

**Pull request merged**